airflow/contrib/hooks/wasb_hook.py

```diff
@@ -36,9 +36,17 @@ class WasbHook(BaseHook):
     :type wasb_conn_id: str
     """
 
-    def __init__(self, wasb_conn_id='wasb_default'):
+    def __init__(self, wasb_conn_id='wasb_default', **kwargs):
         self.conn_id = wasb_conn_id
-        self.connection = self.get_conn()
+        if 'sas_token' in kwargs and kwargs['sas_token'] is not None:
+            self.connection = self.get_custom_conn(**kwargs)
+        else:
+            self.connection = self.get_conn()
```

Comment:
Why do you need this when there's already a mechanism for passing kwargs to the BlockBlobService.

What is special about `sas_token` -- and if there are other values passed here they are just sliently ignored if that is false which doesn't seem very good behaviour.

I think a better approach would be to remove this if/else, to pass all kwargs to the existing `get_conn()` and have that mergbe the kwargs with the extra_dejson from the connection.